packages/commonjs/src/index.js

```diff
@@ -110,7 +108,7 @@ export default function commonjs(options = {}) {
       ast
     );
 
-    setIsCjsPromise(id, isEsModule ? false : Boolean(transformed));
+    setIsCjsPromise(id, isEsModule || isCompiledEsModule ? false : Boolean(transformed));
```

Comment:
This avoids creating a proxy module for these, which I think is correct. I'm not entirely sure if this creates other unintended side effects?

Reply:
I think that if you are reconstructing as an es module, and that es module will still have `require` calls with dynamic arguments, then the transformation should still take place - after the reconstruction.